src/openfl/text/TextField.hx

```diff
@@ -223,6 +224,23 @@ class TextField extends InteractiveObject implements IShaderDrawable {
 		
 		if (charIndex < 0 || charIndex > __text.length - 1) return null;
 		
+		var rect = new Rectangle ();
+		
+		if (__getCharBoundaries (charIndex, rect)) {
+			
+			return rect;
+			
+		} else {
+			
+			return null;
+			
+		}
+		
+	}
+	
+	
+	function __getCharBoundaries (charIndex:Int, rect:Rectangle):Bool {
```

Comment:
IMHO name of the method `__getCharBoundaries` is not correct any more as it returns only success/failure boolean flag instead of boundaries. Name like '_find/_calc/_fill/(or similar)CharBoundaries` would fit better

Reply:
... Or even better, as I can see it seems that there will be no char boundaries only if there is no `layoutGroups`, namely TextField has no text. Therefore, we can have a separate method `_hasCharBoundaries/_hasText/(or similar)` and keep `__getCharBoundaries` as it is. This way we will save one rectangle object allocation (see line 227) and the code will be better

Reply:
I think it's (kind of) fine, because:
 - it's consistent with current upstream (not like we care anymore but still)
 - it's consistent with e.g. `__getBounds` (which can also be debated, but still)
 - conceptually, it still "returns" a rectangle, just using the "out argument" instead of `return`.

That said, on the other hand I also like "calculate" term more even if it would properly return a Rectangle, because that's what it actually does: not just retrieving a value but actually calculating/creating one.

Reply:
So, `__calculateCharBoundaries`?

Reply:
> we can have a separate method `_hasCharBoundaries/_hasText/(or similar)` and keep `__getCharBoundaries` as it is. This way we will save one rectangle object allocation (see line 227) and the code will be better

Good point! Will look into this. Though I still think "calculate" is a better term.

Reply:
Ok, this change is obsolete, looks like I won't touch `getCharBoundaries` within the scope of this PR :)